definitions/procedures/puppet/disable_puppet.rb

```diff
@@ -0,0 +1,46 @@
+module Procedures::Puppet
+  class DisablePuppet < ForemanMaintain::Procedure
+    metadata do
+      description 'Disable Puppet feature'
+      confine do
+        feature(:puppet_server)
+      end
+    end
+
+    def run
+      execute!('foreman-rake db:migrate VERSION=0 SCOPE=foreman_puppet') if server_with_puppet?
+      feature(:installer).run(installer_arguments_disabling_puppet.join(' '), :interactive => false)
+      packages_action(:remove, packages_to_remove, :assumeyes => true)
+    end
+
+    private
+
+    def server_with_puppet?
+      find_package(foreman_plugin_name('foreman_puppet'))
+    end
+
+    def installer_arguments_disabling_puppet
+      options = [
+        '--no-enable-puppet',
+        '--puppet-server false',
+        '--foreman-proxy-puppet false',
+        '--foreman-proxy-puppetca false',
+        '--foreman-proxy-content-puppet false'
+      ]
+      if server_with_puppet_plugin?
+        options << '--no-enable-foreman-plugin-puppet'
+        options << '--no-enable-foreman-cli-puppet'
+      end
+      options
```

Comment:
This is where I'm a bit limited on my knowledge, but going by `definitions/features/katello.rb` we can have use `feature(:installer).answers` to get the installer answers.

From there on we can detect what's available. So this would make it fully dynamic. Technically the check if `puppet` is present is redundant since it's always present:

```suggestion
    def installer_arguments_disabling_puppet
      answers = feature(:installer).answers

      options = []
      options << '--no-enable-puppet' if answers.key?('puppet')
      options << '--no-enable-foreman-plugin-puppet' if answers.key?('foreman::plugin::puppet')
      options << '--no-enable-foreman-cli-puppet' if answers.key?('foreman::cli::puppet')
      if answers['foreman_proxy']
        options << '--foreman-proxy-puppet false'
        options << '--foreman-proxy-puppetca false'
      end
      options << '--foreman-proxy-content-puppet false' if answers.key?('foreman_proxy_content')
      options
    end
```